bindings/csharp/Ckzg.Bindings/Ckzg.cs

```diff
@@ -28,58 +51,65 @@ public class Ckzg
+    /// <param name="commitment_bytes"></param>
+    /// <param name="z_bytes"></param>
+    /// <param name="y_bytes"></param>
+    /// <param name="proof_bytes"></param>
     /// <param name="ts">Trusted setup settings</param>
-    /// <returns>Returns error code or <c>0</c> if the proof is correct</returns>
-    [DllImport("ckzg", EntryPoint = "verify_kzg_proof_wrap", CallingConvention = CallingConvention.Cdecl)]
-    public unsafe static extern int VerifyKzgProof(byte* commitment_bytes, byte* z_bytes, byte* y_bytes, byte* proof_bytes, IntPtr ts);
+    /// <returns></returns>
+    [DllImport("ckzg", EntryPoint = "verify_kzg_proof", CallingConvention = CallingConvention.Cdecl)]
+    public unsafe static extern Ret VerifyKzgProof(bool* result, byte* commitment_bytes, byte* z_bytes, byte* y_bytes, byte* proof_bytes, IntPtr ts);
 
     /// <summary>
-    /// Load trusted setup settings from file
+    /// Given a blob and a KZG proof, verify that the blob data corresponds to the provided commitment.
     /// </summary>
-    /// <param name="filename">Settings file path</param>
-    /// <returns>Trusted setup settings as a pointer or <c>0</c> in case of failure</returns>
-    [DllImport("ckzg", EntryPoint = "load_trusted_setup_wrap")] // free result with free_trusted_setup()
-    public static extern IntPtr LoadTrustedSetup(string filename);
+    /// <param name="result">True if the proof is valid</param>
+    /// <param name="blob"></param>
+    /// <param name="commitment"></param>
+    /// <param name="proof"></param>
+    /// <param name="ts">Trusted setup settings</param>
+    /// <returns></returns>
+    [DllImport("ckzg", EntryPoint = "verify_blob_kzg_proof", CallingConvention = CallingConvention.Cdecl)]
+    public unsafe static extern Ret VerifyBlobKzgProof(bool* result, byte* blob, byte* commitment_bytes, byte* proof_bytes, IntPtr ts);
 
     /// <summary>
-    /// Frees memory allocated for trusted setup settings
+    /// Given a list of blobs and blob KZG proofs, verify that they correspond to the provided commitments.
     /// </summary>
+    /// <param name="result">True if the proofs are valid</param>
+    /// <param name="blobs">Blobs as a flattened byte array</param>
+    /// <param name="commitments">Commitments as a flattened byte array</param>
+    /// <param name="proofs">Proofs as a flattened byte array</param>
+    /// <param name="count">The number of blobs/commitments/proofs</param>
     /// <param name="ts">Trusted setup settings</param>
-    [DllImport("ckzg", EntryPoint = "free_trusted_setup_wrap", CallingConvention = CallingConvention.Cdecl)]
-    public static extern void FreeTrustedSetup(IntPtr ts);
+    /// <returns></returns>
+    [DllImport("ckzg", EntryPoint = "verify_blob_kzg_proof_batch", CallingConvention = CallingConvention.Cdecl)]
+    public unsafe static extern Ret VerifyBlobKzgProofBatch(bool* result, byte* blobs, byte* commitments_bytes, byte* proofs_bytes, int count, IntPtr ts);
```

Comment:
I'm surprised by how thin this binding is compared to other similar bindings. For example [the Rust bindings](https://github.com/ethereum/c-kzg-4844/blob/15101c3dd55026eef7f026b0fb24f216e0381db3/bindings/rust/src/lib.rs#L256) which do explicit length checks (or the Java bindings which do even more checks).

Am I missing the place where the length checks happen? Or are we inconsistent between the two bindings? If so, we should figure out which one is right.

Reply:
I think we should make these changes in a subsequent PR, this one has already changed a lot. And I'd prefer if I weren't the one to do this, but I will if no one wants to.

Reply:
It's was delegated to a consumer library, I can bring that checks from there, diffidently a good idea to have them here.

Reply:
Yes please! :)